adoc/code/graph.cpp

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2025 The Khronos Group, Inc.
+// SPDX-License-Identifier: MIT
+
+#include <iostream>
+#include <sycl/sycl.hpp>
+#include <utility>
+using namespace sycl;  // (optional) avoids need for "sycl::" before SYCL names
+
+constexpr size_t N = 100;
+constexpr size_t Iter = 10;
+
+int main() {
+  // Create a queue to work on.
+  queue myQueue({property::queue::in_order{}});
+
+  // Create some 1D arrays of float
+  float* AOld = malloc_device<float>(N, myQueue);
+  float* ANew = malloc_device<float>(N, myQueue);
+
+  khr::command_graph myGraph{myQueue};
+
+  myGraph.begin_recording(myQueue);
+
+  // Record an asynchronous kernel to compute even iterations
+  myQueue.submit([&](handler& cgh) {
```

Comment:
Having never seen this code or algorithm, it's not immediately clear to me what the example is trying to compute. I think trying to understand it detracts from understanding the graph API. Would it be possible to add some comments that explain what is being computed and what the expected results are?

Reply:
Yes, good point. That's exactly what I was trying to avoid.

Let me add some details on the common pattern that the example mimics, and why we have to record two iterations.

Reply:
It could help to have a more detailed description at the beginning of the sample which describes what the algorithm is doing.

Reply:
I've added a brief description at the beginning. 

Brief, because the focus should be on the graph recording and replay from my perspective.